### PR TITLE
Bug 1832305: Kuryr: Remove old SG rules on upgrade

### PR DESCRIPTION
**Pull request by dulek**
In order to support updates in the SG rules added on Kuryr bootstrap we
need to make sure the security groups that got replaced by others are
removed. This commit, by maintaing list of "decommissioned" SG rules,
will make sure we delete them on upgrade.

### CODE REVIEW

File: pkg/platform/openstack/networking.go [hunk 1]
```diff
@@ -366,3 +366,17 @@ func ensureOpenStackSgRule(client *gophercloud.ServiceClient, sgId, remotePrefix
 	}
 	return nil
 }
+
+// Returns list of OpenStack ingress security group rules on sgId SG.
+func listOpenStackSgRules(client *gophercloud.ServiceClient) ([]rules.SecGroupRule, error) {
+	page, err := rules.List(client, rules.ListOpts{Direction: "ingress"}).AllPages()
```
Comment (luis5tb) [inline pkg/platform/openstack/networking.go:372]: what about the tag? and if this is only for master/workers, why not restrict the search to those SGs?

File: pkg/platform/openstack/kuryr_bootstrap.go [hunk 2]
```diff
@@ -376,11 +376,16 @@ func BootstrapKuryr(conf *operv1.NetworkSpec, kubeClient client.Client) (*bootst
...
 	var sgRules = []sgRule{
-		{podSgId, "0.0.0.0/0", -1, -1, rules.ProtocolTCP},
+		{podSgId, "0.0.0.0/0", 0, 0, rules.ProtocolTCP},
 		{masterSgId, openStackSvcCIDR, etcdPort, etcdPort, rules.ProtocolTCP},
 		{masterSgId, openStackSvcCIDR, apiPort, apiPort, rules.ProtocolTCP},
 	}
 
+	var decommissionedRules = []sgRule{
+		{podSgId, workerSubnet.CIDR, 0, 0, ""},
+		{masterSgId, openStackSvcCIDR, 2379, 2380, rules.ProtocolTCP},
```
Comment (luis5tb) [inline pkg/platform/openstack/kuryr_bootstrap.go:386]: aren't these the etcd ports? shouldn't we remove other ranges not this one?

File: pkg/platform/openstack/kuryr_bootstrap.go [hunk 3]
```diff
@@ -376,11 +376,16 @@ func BootstrapKuryr(conf *operv1.NetworkSpec, kubeClient client.Client) (*bootst
...
 
 	var sgRules = []sgRule{
-		{podSgId, "0.0.0.0/0", -1, -1, rules.ProtocolTCP},
+		{podSgId, "0.0.0.0/0", 0, 0, rules.ProtocolTCP},
 		{masterSgId, openStackSvcCIDR, etcdPort, etcdPort, rules.ProtocolTCP},
 		{masterSgId, openStackSvcCIDR, apiPort, apiPort, rules.ProtocolTCP},
 	}
 
+	var decommissionedRules = []sgRule{
+		{podSgId, workerSubnet.CIDR, 0, 0, ""},
```
Comment (luis5tb) [inline pkg/platform/openstack/kuryr_bootstrap.go:385]: shouldn't be the other way around? removing the access from the pods to the workers?

File: pkg/platform/openstack/networking.go (hunk 1 above)
Comment (dulek) [reply pkg/platform/openstack/networking.go:372]: Nice idea!

File: pkg/platform/openstack/kuryr_bootstrap.go (hunk 3 above)
Comment (dulek) [reply pkg/platform/openstack/kuryr_bootstrap.go:385]: This is supposed to remove [this rule](https://github.com/openshift/cluster-network-operator/pull/465/files#diff-4ab6aa530ac077ca991bba003d6ea267L1016) that got replaced by [this rule](https://github.com/openshift/cluster-network-operator/pull/465/files#diff-4ab6aa530ac077ca991bba003d6ea267R1017).

File: pkg/platform/openstack/kuryr_bootstrap.go (hunk 2 above)
Comment (dulek) [reply pkg/platform/openstack/kuryr_bootstrap.go:386]: [This rule from 4.3](https://github.com/openshift/cluster-network-operator/pull/465/files#diff-4ab6aa530ac077ca991bba003d6ea267L1035) got replaced by a [rule opening just 2379 in 4.4](https://github.com/openshift/cluster-network-operator/pull/465/files#diff-4ab6aa530ac077ca991bba003d6ea267R1043).

File: pkg/platform/openstack/networking.go (hunk 1 above)
Comment (dulek) [reply pkg/platform/openstack/networking.go:372]: Oh crap, seems like we don't tag pod SG yet…

Comment (dulek) [reply pkg/platform/openstack/networking.go:372]: Nevermind.

File: pkg/platform/openstack/kuryr_bootstrap.go (hunk 2 above)
Comment (luis5tb) [reply pkg/platform/openstack/kuryr_bootstrap.go:386]: right!

File: pkg/platform/openstack/kuryr_bootstrap.go (hunk 3 above)
Comment (luis5tb) [reply pkg/platform/openstack/kuryr_bootstrap.go:385]: ok

**Comment by MaysaMacedo** [comment]
LGTM.

**Pull request merged**